lib/vault.js

```diff
@@ -0,0 +1,56 @@
+const { exec } = require('child_process');
+const fs = require('fs');
+
+const service = process.env.SERVICE_SYSTEM_CODE;
+let region = (process.env.REGION || 'LOCAL').toLowerCase();
+const vaultPath = `secret/teams/origami/${service}/${region}`;
+const envFile = '.env';
+
+if (!service) {
+    console.warn(
+        'Could not load enviroment variables from Vault. Service name is undefined.',
+        'Set the enviroment variable "SERVICE_SYSTEM_CODE" in your Makefile.'
+    );
+    process.exit(1);
+}
+
+// Update an existing .env file with enviroment variables from Vault.
+exec(`vault read -format=json ${vaultPath}`, (err, stdout, stderr) => {
+    const result = err ? null : JSON.parse(stdout);
+
+    // Handle errors.
+    if (err || !result.data) {
+        console.warn(
+            `Could not load enviroment variables from Vault (${vaultPath}).`,
+            (stdout ? `\nstdout: ${stdout}` : ''),
+            (stderr ? `\nstderr: ${stderr}` : '')
+        );
+        process.exit(1);
+    }
+
+    // Get secrets object.
+    var existingSecrets = fs.readFileSync(envFile).toString().trim().split('\n');
+    existingSecrets = existingSecrets.reduce((secretsObject, secret) => {
+        const secretArr = (secret.includes('=') ? secret.split('=') : null);
+        if (secretArr) {
+            secretsObject[secretArr[0]] = secretArr[1];
+        }
+        return secretsObject;
+    }, {});
+    const vaultSecrets = result.data;
+    const secrets = Object.assign(existingSecrets, vaultSecrets)
```

Comment:
This would overwrite an existing secret with a secret of the same name from vault, is this the expected behaviour?

Reply:
That's what I intended, expecting Vault to be the source of truth. I did not want to remove an environment variable not configured in vault though.

Reply:
I think this is what we want, yeah. One addition that might be useful though is if the script warns when a secret is found locally but not in Vault – there's a chance it _should_ be in Vault.

Reply:
Good suggestion, I've added that.